new-new-packages/compiler/src/compiler/hir.rs

```diff
@@ -120,8 +125,14 @@ pub enum Expression {
     Struct(HashMap<Id, Id>),
     Lambda(Lambda),
     Body(Body),
-    Call { function: Id, arguments: Vec<Id> },
-    Error,
+    Call {
+        function: Id,
+        arguments: Vec<Id>,
+    },
+    Error {
+        child: Option<Id>,
+        errors: Vec<CompilerError>,
```

Comment:
I think it always contains just a single error, so the vec is unnecessary

Reply:
It can contain multiple errors, in particular multiple errors from lower levels.
For example, if a non-identifier is called, an `AstError::CallOfANonIdentifier` is generated, but all errors of the arguments are collected as well and returned in the same error node.